src/java.base/share/classes/java/util/Timer.java

```diff
@@ -537,6 +565,18 @@ private void mainLoop() {
                             continue;  // No action required, poll queue again
                         }
                         currentTime = System.currentTimeMillis();
+                        /*[IF CRIU_SUPPORT]*/
+                        if (task.criuAdjustRequired) {
+                            long checkpointRestoreTimeDelta = InternalCRIUSupport.getCheckpointRestoreNanoTimeDelta();
+                            // A zero checkpointRestoreTimeDelta value indicates no Checkpoint performed yet,
+                            // it can't be negative, otherwise a RestoreException already was thrown.
+                            if (checkpointRestoreTimeDelta > 0) {
```

Comment:
since we ensure that it cant be negative, the ` if (checkpointRestoreTimeDelta > 0) {` is no longer needed.

Reply:
It could be zero though, i.e., no Checkpoint performed yet, in that case, `task.criuAdjustRequired` can't be cleared (set to `false`) particularly for repeating scheduled task.

Reply:
ah okay